### ISSUE TITLE
Unlimited invoking of fetchMoreItems when scrolling in useInfiniteLoader

### ISSUE THREAD

**Issue by soly2014** [jaredLunde/masonic#42]
`  const fetchMoreItems = (startIndex, stopIndex) => {
      dispatch(fetchProductVariantsStart());
  };
`
i have unlimited dispatching of redux actions when scrolling the page

**Comment by jaredLunde** [comment]
Masonic doesn't care how you load more items. Can be asynchronous, synchronous, whatever. If it needs more items it's going to keep asking for them. Thus, you are in charge of providing an `isItemLoaded` option and memoizing the `loadMoreItems` callback if you need to.

**Comment by soly2014** [comment]
this is the code i use
 ` 
const fetchMoreItems = (startIndex, stopIndex) => { dispatch(fetchProductVariantsStart()); };

const maybeLoadMore = useInfiniteLoader(fetchMoreItems, {
    isItemLoaded: (index, items) => !!items[index],
    minimumBatchSize: 10,
    threshold: 3,
  });
`
if you can help me i would be gratefull

**Comment by bjenkins24** [comment]
@soly2014 did you ever figure it out? I seem to be having a similar issue. The problem seems to be that `onRender` is not throttled and the state can't update fast enough to only make a single asynchronous call. When `onRender` is called my async request in `loadMoreItems` kicks off. Items isn't update until that request finishes. In this case I did a `isLoading` param and did a `useCallback` for the `loadMoreItems` function.

It's now cut it down to two requests per page. But it does two I'm assuming because `isLoading` is not set as fast as `onRender` can render. I'd like to get it down to one request. Tried throttling/debouncing myself, but couldn't quite get it there.

I was only successful getting to one request by unmounting and mounting the entire `Masonic` component while the async request is loading, which is obviously not going to work.

@jaredLunde any guidance here?
